recipe/meta.yaml

```diff
@@ -42,7 +42,7 @@ outputs:
     files:
       - lib/libopenblasp*{{ SHLIB_EXT }}   # [not win]
       - lib/libopenblas.so.*               # [linux]
-      - lib/libopenblas.*.dylib            # [osx]
+      - lib/libopenblas.*dylib             # [osx]
```

Comment:
Why? No downstream package is linking to `libopenblas.dylib`

Reply:
I'm not sure you are fully aware of how compiler share object resolution works.

see https://www.ibm.com/developerworks/linux/library/l-shlibs/index.html
Compatibility's not just for relationships 

This is how most shared objects work in *nix

Reply:
If you look at the original issue you can see the _multiarray_umath.so references it

Reply:
I'm aware. Can you share the conda environment and the exact commands you use that makes this an issue?

Reply:
Original issue says,
```
  Referenced from: /Users/jiahuanglin/anaconda3/envs/test/lib/python2.7/site-packages/numpy/core/_multiarray_umath.so
  Reason: image not found
```

There's no openblas mentioned anywhere. I suspect the user is using `numpy` from `defaults` and `libopenblas` from `conda-forge`. That's why I asked for the conda environment information

Reply:
I'ved asked the OP to list it, but the original build.sh is clear about how the linkage works:

```
    # Needs to fix the install name of the dylib so that the downstream projects will link
    # to libopenblas.dylib instead of libopenblasp-r0.2.20.dylib
    # In linux, SONAME is libopenblas.so.0 instead of libopenblasp-r0.2.20.so, so no change needed
```

Reply:
I've changed that to `libopenblas.0.dylib` in openblas 0.3.6 and the comment has to be updated. I've also sent a patch upstream to fix it, https://github.com/xianyi/OpenBLAS/pull/2181

Reply:
Ah OK, I can see that:

numpy from defaults:
```
otool -L ~/conda/envs/test2/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-darwin.so
/Users/steve/conda/envs/test2/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-darwin.so:
    @rpath/libopenblas.dylib (compatibility version 0.0.0, current version 0.0.0)
    /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
```

numpy from conda-forge:
```
otool -L ~/conda/envs/test/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-darwin.so
/Users/steve/conda/envs/test/lib/python3.7/site-packages/numpy/core/_multiarray_umath.cpython-37m-darwin.so:
    @rpath/libcblas.3.dylib (compatibility version 0.0.0, current version 0.0.0)
    /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```

Why was the unversioned dylib deprecated? The industry standard is to keep the symlink to avoid causing breakage like this. I don't see the rationale in https://github.com/conda-forge/openblas-feedstock/pull/70#discussion_r288424931

Reply:
Because none of the packages in conda-forge link to libopenblas.dylib, and in defaults each package linking to libopenblas.dylib pins openblas to exact version. So, when we updated to 0.3.6, I thought defaults would use the same script as ours, but they didn't. I finally got them to accept it https://github.com/AnacondaRecipes/openblas-feedstock/pull/3